src/core/Table/filters/TextFilter/TextFilter.tsx

```diff
@@ -0,0 +1,51 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import React from 'react';
+import '@itwin/itwinui-css/css/table.css';
+import { useTheme } from '../../../utils/hooks/useTheme';
+import { Input } from '../../../Input';
+import {
+  FilterButtonBar,
+  FilterButtonBarLocalization,
+} from '../FilterButtonBar';
+import { TableFilterProps } from '../types';
+
+export type TextFilterProps<
+  T extends Record<string, unknown>
+> = TableFilterProps<T> & {
+  localization?: FilterButtonBarLocalization;
+};
+
+export const TextFilter = <T extends Record<string, unknown>>(
+  props: TextFilterProps<T>,
+) => {
+  const { column, setFilter, clearFilter, localization } = props;
+
+  useTheme();
+
+  const [text, setText] = React.useState(column.filterValue ?? '');
+
+  const onKeyDown = (event: React.KeyboardEvent<HTMLDivElement>) => {
```

Comment:
hmm...
Putting whole thing into form and having a submit button wouldnt take care of that, I wonder...

Reply:
We both tested and it not really a better way.